apps/designer/app/canvas/shared/styles.ts

```diff
@@ -225,16 +225,26 @@ const useUpdateTokens = () => {
 
 const usePreviewStyle = () => {
   useSubscribe("previewStyle", ({ id, updates, breakpoint }) => {
-    if (getRule(id, breakpoint.id) === undefined) {
-      const rule = addRule(id, { breakpoint: breakpoint.id, style: {} });
-      for (const update of updates) {
-        rule.styleMap.set(update.property, update.value);
-      }
-      cssEngine.render();
+    let rule = getRule(id, breakpoint.id);
+
+    if (rule === undefined) {
+      rule = addRule(id, { breakpoint: breakpoint.id, style: {} });
     }
+
     for (const update of updates) {
+      // This is possible on newly created instances, properties are not yet defined in the style.
+      if (rule.styleMap.get(update.property) == null) {
```

Comment:
== null is confusing, it will rever ever return a null, its only undefined

Reply:
You make it look like the api could return null or undefined

Reply:
Switched on has, as basically we are checking existance, also fixed PlaintextRule styleMap on  `new StylePropertyMap();`
https://github.com/webstudio-is/webstudio-designer/pull/551/commits/274e68843d39fa70b9492576363b817be01e049f
as it cause `any` in the return type of `get`